ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -373,6 +391,57 @@ void fillChunks() {
   }
 
   /**
+   * Submits encrypt job for the given {@link PutChunk} and processes the callback for the same
+   * @param putChunk the {@link PutChunk} that needs to be encrypted
+   * @param encryptContent {@code true} if content needs to be encrypted. {@code false} otherwise
+   */
+  private void encryptChunk(PutChunk putChunk, boolean encryptContent) {
+    try {
+      logger.trace("Chunk at index {} state moves to {}", putChunk.chunkIndex, ChunkState.Encrypting);
+      putChunk.state = ChunkState.Encrypting;
+      if (encryptContent) {
+        putChunk.buf.flip();
+        putChunk.chunkBlobSize = putChunk.buf.remaining();
+      }
+      logger.trace("Submitting encrypt job for chunk at index " + putChunk.chunkIndex);
+      exec.submitJob(new EncryptJob(passedInBlobProperties.getAccountId(), passedInBlobProperties.getContainerId(),
```

Comment:
this is too long a statement I think

Reply:
yes, unfortunately EncryptJob has too many arguments. If you can think of a way to simplify things, let me know.